src/main/java/dk/aau/cs/model/CPN/ColorType.java

```diff
@@ -89,6 +97,18 @@ public Color getFirstColor() {
         return colors.firstElement();
     }
 
+    public Color getLastColor() {
+        return colors.lastElement();
+    }
+
+    private String getFirstColorName() {
+        return getFirstColor().getColorName();
+    }
+
+    private String getLastColorName() {
+        return getLastColor().getColorName();
+    }
+
```

Comment:
I would suggest not to add these methods and move this directly to the if-condition where it is used.